src/main/java/net/atos/entng/support/export/TicketExportWorker.java

```diff
@@ -0,0 +1,184 @@
+    private TimelineHelper timelineHelper;
+    private FileService fileService;
+    private WorkspaceService workspaceService;
+
+    protected String exportNotification;
+
+    @Override
+    public void start() {
+        super.start();
+        ticketServiceSql = new TicketServiceSqlImpl(null);
+        TicketServiceNeo4jImpl ticketServiceNeo4jImpl = new TicketServiceNeo4jImpl();
+        ticketService = new TicketServiceImpl(ticketServiceNeo4jImpl);
+
+        Storage storage = new StorageFactory(vertx, new JsonObject()).getStorage();
+        workspaceService = new DefaultWorkspaceService(this.vertx, storage, config());
+        fileService = new DefaultFileService(storage);
+        workspaceHelper = new WorkspaceHelper(vertx.eventBus(), storage);
+        timelineHelper = new TimelineHelper(this.vertx, this.vertx.eventBus(), config());
+
+        exportContext = vertx.getOrCreateContext();
+        String launchLog = String.format("[Common@%s::start] Launching worker %s, deploy verticle %s",
+                this.getClass().getSimpleName(), this.getClass().getSimpleName(), exportContext.deploymentID());
+        log.info(launchLog);
+
+        eb.consumer(this.getClass().getName(), this);
+    }
+
+    @Override
+    public void handle(Message<JsonObject> event) {
+        event.reply(new JsonObject().put("status", "ok"));
+        JsonObject params = event.body();
+        UserInfos user = UserInfosHelper.getUserInfosFromJSON(event.body().getJsonObject(Ticket.USER));
+
+        log.info(String.format("[Common@%s::run] Starting worker %s process, generating export file",
+                this.getClass().getSimpleName(), this.getClass().getSimpleName()));
+
+        String local = params.getString(Ticket.LOCALE);
+        String domain = params.getString(Ticket.DOMAIN);
+        I18nConfig i18nConfig = new I18nConfig(domain, local);
+
+        this.exportNotification = "support.export-events";
+        createCSVFile(params, i18nConfig)
+                .compose(file -> exportData(file, user, i18nConfig))
+                .compose(fileInfos -> sendNotification(user, fileInfos))
+                .onFailure(fail -> {
+                    exportNotification = "support.export-events-error";
+                    sendNotification(user, new JsonObject());
+                });
+    }
+
+    private Future<ExportFile> createCSVFile(JsonObject params, I18nConfig i18nConfig) {
+        Promise<ExportFile> promise = Promise.promise();
+        String structureId = params.getString(Ticket.STRUCTURE_ID);
+
+        Future<JsonObject> structureIdsFuture;
+        if (Objects.equals(structureId, Ticket.ASTERISK))
+            structureIdsFuture = Future.succeededFuture(new JsonObject().put(Ticket.STRUCTUREIDS,
+                    params.getJsonObject(Ticket.USER).getJsonArray(Ticket.STRUCTURES)));
+        else
+            structureIdsFuture = ticketService.listStructureChildren(structureId);
+
+        structureIdsFuture
```

Comment:
En vrai, quand tu as ces cas de figure là, l'idéal c'est de faire une méthode à part pour bien distinguer les traitements:
```suggestion
private Future<JsonObject> getStructureIds(String structureId, JsonObject params) {
    if (Objects.equals(structureId, Ticket.ASTERISK))
            return Future.succeededFuture(new JsonObject().put(Ticket.STRUCTUREIDS,
                    params.getJsonObject(Ticket.USER).getJsonArray(Ticket.STRUCTURES)));
    return ticketService.listStructureChildren(structureId);
}
private Future<ExportFile> createCSVFile(JsonObject params, I18nConfig i18nConfig) {
        ...
       getStructureIds(structureId, params)
               .compose(structureIds -> ticketServiceSql.getTicketsFromArrayOfStructureId(structureIds))
               ...
```